src/v/kafka/data/cloud_topic_partition_reader.cc

```diff
@@ -103,6 +103,19 @@ cloud_topic_partition_reader_impl::maybe_load_slices_from_cache() {
             // from S3.
             break;
         }
+        vlog(
+          kdlog.trace,
+          "Loaded batch from cache: {}",
+          batch.value().base_offset(),
+          batch.value().term());
+        if (batch.value().term() == model::term_id{-1}) {
+            // Skip bogus batch
+            vlog(
+              kdlog.warn,
+              "Batch with the invalid term is loaded: {}",
+              batch.value());
+            break;
+        }
```

Comment:
When can this happen? Is this defensive - if so should it be a `vassert` or error log?